vm/ubpf_vm.c

```diff
@@ -45,20 +45,28 @@ void ubpf_set_error_print(struct ubpf_vm *vm, int (*error_printf)(FILE* stream,
 }
 
 struct ubpf_vm *
-ubpf_create(void)
+ubpf_create(void *zmalloc_cookie,
+            ubpf_zmalloc_fn zmalloc_fn,
+            ubpf_free_fn free_fn)
 {
-    struct ubpf_vm *vm = calloc(1, sizeof(*vm));
+    struct ubpf_vm *vm = zmalloc_fn(zmalloc_cookie, sizeof(*vm));
     if (vm == NULL) {
         return NULL;
     }
 
-    vm->ext_funcs = calloc(MAX_EXT_FUNCS, sizeof(*vm->ext_funcs));
+    vm->zmalloc_cookie = zmalloc_cookie;
+    vm->zmalloc_fn = zmalloc_fn;
```

Comment:
Why not create a few sample zmalloc_fn's, and if the passed in function pointers are zero, then use the local functions (which could use calloc/free for example)

I'm thinking we're the only user ... but just in case, do you think it makes sense to create a ubpf_create_XXX() which has the zmalloc/free fn's, and leave the old ubpf_create() as is, and use that to call ubpf_create_XXX(...) with zero's, which would cause it to use the local versions of the functions?

Reply:
Reintroduced ubpf_create() which uses default allocator and deallocator functions.